Doc/library/tempfile.rst

```diff
@@ -75,20 +75,36 @@ The module defines the following user-callable items:
       Added *errors* parameter.
 
 
-.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None)
-
-   This function operates exactly as :func:`TemporaryFile` does, except that
-   the file is guaranteed to have a visible name in the file system (on
-   Unix, the directory entry is not unlinked).  That name can be retrieved
-   from the :attr:`name` attribute of the returned
-   file-like object.  Whether the name can be
-   used to open the file a second time, while the named temporary file is
-   still open, varies across platforms (it can be so used on Unix; it cannot
-   on Windows).  If *delete* is true (the default), the file is
-   deleted as soon as it is closed.
+.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None, delete_on_close=True)
+
+   This function operates exactly as :func:`TemporaryFile` does, except the
+   following differences:
+
+   * The file is guaranteed to have a visible name in the file system (on Unix, the directory entry is not unlinked).
+
+   * There is more granularity in the deletion behaviour of the file
+     (see ``delete_on_close`` below)
+
    The returned object is always a file-like object whose :attr:`!file`
-   attribute is the underlying true file object. This file-like object can
-   be used in a :keyword:`with` statement, just like a normal file.
+   attribute is the underlying true file object. This file-like object can be
+   used in a :keyword:`with` statement, just like a normal file.
+
+   The name of the temporary file can be retrieved from the :attr:`name`
+   attribute of the returned file-like object. Whether the name can be used to
+   open the file a second time, while the named temporary file is still open,
+   varies across platforms (it can be so used on Unix; it cannot on Windows).
+
+   If *delete* is true (the default) and *delete_on_close* is true (the
+   default), the file is deleted as soon as it is closed. If *delete* is true
+   and *delete_on_close* is false, the file is deleted on context manager exit
```

Comment:
In Windows, if `delete_on_close` is false, and the file is created in a directory for which the user lacks delete access (either directly for files as an inherited rule that grants `DELETE` access, or at the directory level via `FILE_DELETE_CHILD` access), then the `unlink()` call on exit of the context manager will fail with a `PermissionError` due to an OS access-denied error (5). This cannot happen when `delete_on_close` is true because `DELETE` access is implicitly requests when `CreateFileW()` is called with `FILE_FLAG_DELETE_ON_CLOSE`, so the open immediately fails if the caller lacks delete access.

Reply:
@eryksun I have implemented your comments. Thanks!
I have also slightly rearranged information. I think the flow is more logical now